pkg/kafka/partitionring/consumer/balancer_test.go

```diff
@@ -0,0 +1,481 @@
+package consumer
+
+import (
+	"context"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/grafana/dskit/ring"
+	"github.com/stretchr/testify/require"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kfake"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/pkg/kmsg"
+)
+
+// Helper types for testing
+type memberUpdate struct {
+	memberID string
+	topics   map[string][]int32
+}
+
+type mockPartitionRing struct {
+	partitionIDs []int32
+}
+
+// Mock implementation of PartitionRing interface
+func (m *mockPartitionRing) PartitionIDs() []int32 {
+	return m.partitionIDs
+}
+
+type mockPartitionRingReader struct {
+	ring *mockPartitionRing
+}
+
+func (m *mockPartitionRingReader) PartitionRing() *ring.PartitionRing {
+	desc := ring.PartitionRingDesc{
+		Partitions: make(map[int32]ring.PartitionDesc),
+	}
+	for _, id := range m.ring.partitionIDs {
+		desc.Partitions[id] = ring.PartitionDesc{
+			Id:     id,
+			State:  ring.PartitionActive,
```

Comment:
Should inactive partitions behave the same as active partitions or as unused partitions?